pkg/convert/tf.go

```diff
@@ -3041,17 +3198,41 @@ func translateModuleSourceCode(
 	}
 	for _, item := range items {
 		if item.moduleCall != nil {
+			leadingTrivia, _ := getTrivia(sources, item.moduleCall.DeclRange, false)
+			packageName, sandbox := sanboxedAnnotation(string(leadingTrivia.Bytes()))
 			moduleCall := item.moduleCall
-			scopes.getOrAddPulumiName("module."+moduleCall.Name, "", "Component")
-
-			// First things first, check if this module has been seen before. If it has, we don't need to
-			// translate it again.
+			moduleName := scopes.getOrAddPulumiName("module."+moduleCall.Name, "", "Component")
 			moduleKey := makeModuleKey(moduleCall)
 
-			if _, has := modules[moduleKey]; !has {
+			if !sandbox {
+				// check if other modules have been marked with @sandbox with the same source
+				// in that case, we consider this module as sandboxed too
+				for _, sandboxModule := range sandboxedModules {
+					if sandboxModule.moduleCall.SourceAddr == moduleCall.SourceAddr {
+						sandbox = true
+						packageName = sandboxModule.packageName
+						break
+					}
+				}
+			}
+
+			if source, isRemote := moduleFromRemoteRegistry(moduleCall); isRemote {
+				key := fmt.Sprintf("%s-%s-%s", moduleCall.Name, source.Package.Namespace, source.Package.Name)
+				// if the module is marked with @sandbox, then we translate it anyways
+				if _, seen := sandboxedModules[key]; !seen && sandbox {
+					state.sandboxedModuleNames[moduleCall.Name] = moduleName
+					sandboxedModules[key] = &SandboxModule{
+						packageName: packageName,
+						moduleCall:  moduleCall,
+					}
+				}
+			}
+
+			// check if this module has been seen before. If it has, we don't need to
+			// translate it again.
```

Comment:
Can you explain a little what this short-circuits? I thought perhaps every module call needs to be translated fresh even if it maps to the similar version and such it's still a separate instance, or am I misunderstanding?

Reply:
I think this comment is for modules that aren't using `pulumi-terraform-module` so that are not downloaded and converted again. I will re-adjust it here

Reply:
Ah ok sorry I should have read closer. So some pre-existing behavior not quite affected.